datastream-server-restli/src/main/java/com/linkedin/datastream/server/DatastreamServer.java

```diff
@@ -234,6 +236,10 @@ public ServerComponentHealthAggregator getServerComponentHealthAggregator() {
     return _serverComponentHealthAggregator;
   }
 
+  public Properties getProperties() {
```

Comment:
Instead of exposing entire properties set. Consider exposing individual property using getter methods.

Example,
instead of

datastreamServer.getProperties().getDomainProperties(CONFIG_CONNECTOR_PREFIX + datastream.getConnectorName()).getProperty(CONFIG_CHECKPOINT_STORE_URL);

do
datastreamServer.getConnectorCheckpointURL(datastream.getConnectorName());

This will abstract the constants and it's hirarchy away.

Reply:
same applies for other properties.

Reply:
In fact, you can simply implement 

public CustomCheckpointProvider getCustomCheckpointProvider(Datastream datastream);

in DatastreamServer and just call that.

By doing so, you implicitly make it checkpoint implementation agnostic.